rust-runtime/aws-smithy-runtime/src/client/retries/strategy/fixed_delay.rs

```diff
@@ -44,18 +44,19 @@ impl RetryStrategy for FixedDelayRetryStrategy {
         cfg: &ConfigBag,
     ) -> Result<ShouldAttempt, BoxError> {
         // Look a the result. If it's OK then we're done; No retry required. Otherwise, we need to inspect it
-        let error = match ctx.output_or_error() {
-            Ok(_) => {
-                tracing::trace!("request succeeded, no retry necessary");
-                return Ok(ShouldAttempt::No);
-            }
-            Err(err) => err,
-        };
+        let output_or_error = ctx.output_or_error().expect(
+            "This must never be called without reaching the point where the result exists.",
+        );
+        if output_or_error.is_ok() {
+            tracing::trace!("request succeeded, no retry necessary");
+            return Ok(ShouldAttempt::No);
```

Comment:
@rcoh suggested in another PR that `ShouldAttempt::No` should have a reason, maybe a `&'static str`, for debugging. I think that's nicer than having a `tracing::trace!` at each site where we return `No`. Then the logging can be centralized (and it should probably be `debug!` instead of `trace!`).